demos/digit-prediction/digits_prediction.nlogo

```diff
@@ -0,0 +1,527 @@
+extensions [ py ]
+
+globals [ path coords_x coords_y color_list digit]
+
+to setup-py
+  clear-all
+  py:setup py:python
+  (py:run
+    "import numpy as np"
+    "from keras.models import load_model"
+  )
+  ;; load model
+  py:run "model = load_model(\"model.h5\")"
+
+  ;; create coords list, useful to make color_list
+  set coords_x (range max-pxcor)
+  set coords_y (range max-pycor)
+  set color_list []
+end
+
+to draw
+  ifelse mouse-down?
+  [
+    ask patch mouse-xcor mouse-ycor
+    [
+      ifelse rubber
+      [
+        set pcolor black
+      ]
+      [
+        set pcolor white
+      ]
+    ]
+  ]
+  [
+    predict
+  ]
+  display
+end
+
+to predict
+  set color_list []
+  foreach coords_x
+  [
+    x -> foreach coords_y
+    [
+      y ->
+        let c [ pcolor / 10 ] of patch x y
+        set color_list lput c color_list
+    ]
+  ]
```

Comment:
The above would be simplified with `map`:

```
py:set "images" map coords_y [ y ->
  map coords_x  [ x ->
    [ pcolor / 10 ] of patch x y
  ]
]
```

Then there's no need for `reshape` in the python. Just do `array = np.array([images])`.

Reply:
Unfortunately I got an error when I pasted this code. 

> MAP expected this input to be a list, but got an anonyous reporter instead

So I didn't change this part of code.

Reply:
Sorry about that! Here's the corrected code:
```
  py:set "images" map [ y ->
    map [ x ->
      [ pcolor / 10 ] of patch x y
    ] coords-x
  ] coords-y
```